### PR TITLE
[Jenkins] Update the release script and Jenkinsfile

### PR DESCRIPTION
**Pull request by firestarman**
This PR is to update the release script and Jenkinsfile for porting release job to NGCC.

- Update the credential IDs to use the new ones on NGCC .
- Change the default value of parameter "REF" to `main` .
- Use string instead of choice for parameter "DEPLOY_TO", to avoid private URL hardcode.
- Create the folder '.zinc' under home before building.

### CODE REVIEW

File: jenkins/Jenkinsfile.release [hunk 1]
```diff
@@ -91,7 +81,7 @@ pipeline {
                         sh 'rm -rf $GNUPGHOME'
                         sh 'gpg --import $GPG_FILE'
                         retry (3) {
-                            sh "bash $JENKINS_ROOT/deploy.sh true false"
+                            sh "bash $JENKINS_ROOT/deploy.sh true"
```
Comment (GaryShen2008) [inline jenkins/Jenkinsfile.release:84]: I think we should keep "fasle", since deploy.sh needs this parameter to check databricks build or not.

Comment (firestarman) [reply jenkins/Jenkinsfile.release]: updated

**Comment by GaryShen2008** [comment]
build

**Comment by firestarman** [comment]
build

**Comment by tgravescs** [comment]
you need to update the Databricks files as well.

**Comment by firestarman** [comment]
> you need to update the Databricks files as well.

We can update for databricks in another PR.

**Pull request merged**